dedupe/datamodel.py

```diff
@@ -181,16 +181,18 @@ def typify_variables(
             )
 
         variable_object = variable_class(definition)
-        assert isinstance(variable_object, FieldVariable)
+        assert isinstance(variable_object, Variable)
 
         variables.append(variable_object)
 
-    only_custom = all(isinstance(v, CustomType) for v in variables)
-    if only_custom:
+    no_blocking_variables = all(
+        isinstance(v, (CustomType, InteractionType)) for v in variables
+    )
+    if no_blocking_variables:
```

Comment:
Perhaps this check should happen in DataModel.__init__, and not in this utility function?

Reply:
it seems of a piece with the "Incorrect variable specification: variable" error?